### PR TITLE
[P1] Fixing continue menu option when game finishes

### PR DESCRIPTION
**Pull request by podarsmarty**
## What are the changes the user will see?
- Continue button will no longer show up if the last and only session game completes

## Why am I making these changes?
- Continue button shows up erroneously if last and only session game completes. If user selects it game freezes/crashes

## What are the changes from a developer perspective?
- UserInfo was not being properly updated in the post end game phase. There were 2 issues:
    1. it would never update for localStorage only runs
    2. for server runs it would try to update UserInfo before the session was cleared

## How to test the changes?
- Start a session making sure it is the only session & make sure it saves. Restart page and continue the run ending it. Once ended the continue button will no longer show up.

## Bug report
- https://github.com/pagefaultgames/pokerogue/issues/4288

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I considered writing automated tests for the issue? took a quick look. doesn't look like project is setup to mock and test logic for phases touching updating user data
- [ ] If I have text, did I make it translatable and add a key in the English locale file(s)?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [ ] Are the changes visual?
  - [ ] Have I provided screenshots/videos of the changes?

### CODE REVIEW

**Comment by flx-sta** [comment]
@podarsmarty mind adding a video of proof?

**Comment by flx-sta** [comment]
Might want to adapt the whole thing to `async/await` at the same time
```ts
/**
   * Attempt to clear session data. After session data is removed, attempt to update user info so the menu updates
   */
  async tryClearSession(scene: BattleScene, slotId: integer): Promise<[success: boolean, newClear: boolean]> {
    let result: [boolean, boolean] = [false, false];

    if (bypassLogin) {
      localStorage.removeItem(`sessionData${slotId ? slotId : ""}_${loggedInUser?.username}`);
      result = [true, true];
    } else {
      const sessionData = this.getSessionSaveData(scene);
      const response = await Utils.apiPost(`savedata/session/clear?slot=${slotId}&trainerId=${this.trainerId}&secretId=${this.secretId}&clientSessionId=${clientSessionId}`, JSON.stringify(sessionData), undefined, true);

      if (response.ok) {
          loggedInUser!.lastSessionSlot = -1; // TODO: is the bang correct?
          localStorage.removeItem(`sessionData${this.scene.sessionSlotId ? this.scene.sessionSlotId : ""}_${loggedInUser?.username}`);
      }

      const jsonResponse = await response.json();

      if (!jsonResponse.error) {
        result = [true, jsonResponse.success ?? false];
      } else {
        if (jsonResponse && jsonResponse.error?.startsWith("session out of date")) {
          this.scene.clearPhaseQueue();
          this.scene.unshiftPhase(new ReloadSessionPhase(this.scene));
        }

        console.error(jsonResponse);
        result = [false, false];
      }
    }

    const updateUserInfoSuccess = await updateUserInfo();

    if (updateUserInfoSuccess !== null && !updateUserInfoSuccess) {
      result = [false, false];
    }

    return result;
  }
  ```

**Comment by flx-sta** [comment]
I've added a PR in your repo to address the `async/await` migration

https://github.com/podarsmarty/pokerogue/pull/1

**Comment by podarsmarty** [comment]
https://github.com/user-attachments/assets/efc0e975-e909-4d21-b9d3-140d4497f12e

**Review by flx-sta — APPROVED** [review]
lgtm

https://github.com/user-attachments/assets/b968dcf1-4144-4025-93de-718a8f9cd333

File: package.json [hunk 1]
```diff
@@ -36,6 +36,7 @@
 		"inquirer": "^11.0.2",
 		"jsdom": "^24.0.0",
 		"lefthook": "^1.6.12",
+		"msw": "^2.4.9",
```
Comment (ben-lear) [inline package.json:39]: unused import?

Comment (flx-sta) [reply package.json:39]: I've added that. It's being used fo rmy tests

**Review by ben-lear — APPROVED** [review]

**Review by DayKev — APPROVED** [review]

**Pull request merged**